unittests/dry_run_trx_tests.cpp

```diff
@@ -0,0 +1,331 @@
+                                transaction_metadata::trx_type type=transaction_metadata::trx_type::input, uint32_t delay_sec=0) {
+      action act;
+      signed_transaction trx;
+
+      act.account = "noauthtable"_n;
+      act.name = name;
+      act.authorization = auth;
+      act.data = data;
+
+      trx.actions.push_back( act );
+      set_transaction_headers( trx );
+      trx.delay_sec = delay_sec;
+      if ( type == transaction_metadata::trx_type::input ) {
+         trx.sign(get_private_key("alice"_n, "active"), control->get_chain_id());
+      }
+
+      return push_transaction( trx, fc::time_point::maximum(), DEFAULT_BILLED_CPU_TIME_US, false, type );
+   }
+
+   void insert_a_record() {
+      auto res = send_db_api_transaction("insert"_n, insert_data);
+      BOOST_CHECK_EQUAL(res->receipt->status, transaction_receipt::executed);
+      produce_block();
+   }
+
+   abi_serializer abi_ser{ json::from_string(test_contracts::no_auth_table_abi()).as<abi_def>(), abi_serializer::create_yield_function(abi_serializer_max_time )};
+   bytes insert_data;
+   bytes getage_data;
+};
+
+BOOST_AUTO_TEST_SUITE(dry_run_trx_tests)
+
+BOOST_FIXTURE_TEST_CASE(require_authorization, dry_run_trx_tester) { try {
+   produce_blocks( 1 );
+
+   action act = {
+      {}, // no authorization provided: vector<permission_level>{{config::system_account_name,config::active_name}},
+      newaccount{
+       .creator  = config::system_account_name,
+       .name     = "alice"_n,
+       .owner    = authority( get_public_key( "alice"_n, "owner" ) ),
+       .active   = authority( get_public_key( "alice"_n, "active" ) )
+      }
+   };
+
+   // dry-run requires authorization
+   BOOST_REQUIRE_THROW(send_action(act, false), tx_no_auths);
+
+   // sign trx with no authorization
+   signed_transaction trx;
+   trx.actions.push_back( act );
+   set_transaction_headers( trx );
+   trx.sign(get_private_key("alice"_n, "active"), control->get_chain_id());
+   BOOST_REQUIRE_THROW(push_transaction( trx, fc::time_point::maximum(), DEFAULT_BILLED_CPU_TIME_US, false, transaction_metadata::trx_type::dry_run ), tx_no_auths);
+} FC_LOG_AND_RETHROW() }
+
+BOOST_FIXTURE_TEST_CASE(newaccount_test, dry_run_trx_tester) { try {
+   produce_blocks( 1 );
+
+   action act = {
+      vector<permission_level>{{config::system_account_name,config::active_name}},
+      newaccount{
+         .creator  = config::system_account_name,
+         .name     = "alice"_n,
+         .owner    = authority( get_public_key( "alice"_n, "owner" ) ),
+         .active   = authority( get_public_key( "alice"_n, "active" ) )
+      }
+   };
+
+   send_action(act, false); // should not throw
+   send_action(act, false); // should not throw
+   send_action(act, true); // should not throw
+   BOOST_CHECK_THROW(control->get_account("alice"_n), fc::exception); // not actually created
```

Comment:
Like this check. Maybe you can wait one block to check to make sure state is really not changed by dry-run.